### DISCUSSION TITLE
[Game] Use of AccessControl

### DISCUSSION THREAD

**Discussion by brozorec** [Fujicracy/fuji-v1#180] [General]
@DaigaroCota @tiagofneto what do you think about the idea to use "@openzeppelin/contracts/access/AccessControl.sol" in the ERC1155 of the game for defining a role like `POINTS_COMPOUNDER_ROLE` and use it instead of the modifier `onlyVault`?
I guess this will make the code more readable and less error prone.

**Comment by tiagofneto** [comment]
We have actually discussed that. As the code gets bigger, we are starting to see more and more permission requirements in the functions.
